python/chat/server.py

```diff
@@ -0,0 +1,36 @@
+from flask import jsonify, request
+import datasource
+
+def add_message(message):
+    return datasource.insert_message(message['handle'], message['body'])
+
+def get_messages(max_id=None):
+    return datasource.query_messages(since=max_id)
+
+def cleanup_messages():
+    before = datasource.record_count()
+    datasource.delete_records()
+    after = datasource.record_count()
+    return before - after
+
+def messages_endpoint():
+    result = None
+    if request.method == 'GET':
+        max_id = request.args.get('max_id', 0)
+        result = get_messages(max_id=max_id)
+    elif request.method == 'POST':
+        payload = request.json or {}
+        result = add_message(payload)
+    elif request.method == 'DELETE':
```

Comment:
the delete endpoint is exposed to clients, meaning that anyone can cleanup all the messages. I like the idea of a scheduled purge, per the cron task that you’re setting up, but I prefer not to have it exposed (even via localhost). What other options do we have for safe deletion?

Reply:
The delete endpoint just deletes _expired_ messages (older than 3 hrs at the moment) so it is 'safe' for anyone to call repeatedly. I originally had the cleanup in a python timer thread, but was struggling to get it to behave correctly on shut down. I think I may be able to modify the nignx config so that DELETE /chat/messages can only be called from 127.0.0.1, though. I'd like to avoid adding authentication to this endpoint to keep it simple, but understand if you feel like it is necessary.

Reply:
Definitely no need for auth. With your explanation, I don’t feel strongly about constraining the delete operation to localhost - feel free to remove the constraint if you’d prefer to keep things neat.